pkg/operator/ceohelpers/bootstrap.go

```diff
@@ -141,7 +139,11 @@ func CheckSafeToScaleCluster(
 		return fmt.Errorf("CheckSafeToScaleCluster unrecognized scaling strategy %q", scalingStrategy)
 	}
 
-	nodeCount := len(operatorStatus.NodeStatuses)
+	node, err := masterNodeLister.List(masterNodeSelector)
```

Comment:
let's remove the entire nodeStatus and node listing logic here, I think the memberHealth and quorum check below should suffice. We can run some more payload tests to ensure we don't break anything.

Reply:
how about the logic of  createNodeCertConfigs which uses the master node listing logic
https://github.com/openshift/cluster-etcd-operator/blob/9afc6795eaac93bf854f00ed03db9c243b2157fb/pkg/operator/etcdcertsigner/etcdcertsignercontroller.go#L293-L338

Reply:
I think that can stay for now, because we need to have the certificates before a member can join the cluster. So using the etcd client won't work when a new node joins the cluster :)